multilinear_extensions/src/expression/monomial.rs

```diff
@@ -28,14 +28,14 @@ impl<E: ExtensionField> Expression<E> {
     fn distribute(&self) -> Vec<Term<Expression<E>, Expression<E>>> {
         match self {
             // only contribute to scalar terms
-            Constant(_) | Challenge(..) | Instance(_) => {
+            Constant(_) | Challenge(..) | InstanceScalar(_) => {
```

Comment:
Found an improper handling for Instance (public io). For we have 2 type
- vec length = 1
- vec length > 1

we should put vec length = 1 into scalar, while make > 1 length put in product part.
This issue was only trigger in this PR, after I try to use monomial term for `table circuit` lookup witness generation